sdk/main/include/ECDSAsecp256k1PrivateKey.h

```diff
@@ -147,28 +144,29 @@ class ECDSAsecp256k1PrivateKey : public PrivateKey
 
 private:
   /**
-   * Create an OpenSSL keypair object from a byte vector representing an ECDSAsecp256k1PrivateKey.
+   * Create an OpenSSL key object from a byte vector representing an ECDSAsecp256k1PrivateKey.
    *
    * @param keyBytes The bytes representing a ECDSAsecp256k1PrivateKey.
    * @return A pointer to a newly created OpenSSL keypair object.
    */
-  static EVP_PKEY* bytesToPKEY(const std::vector<unsigned char>& keyBytes);
+  static std::unique_ptr<EVP_PKEY, void (*)(EVP_PKEY*)> bytesToPKEY(const std::vector<unsigned char>& keyBytes);
```

Comment:
Have you considered using `decltype` for the destructor, rather than explicitly typing the type out?

Reply:
No I didn't consider that, can add it though. I think I'll add an alias for these classes as well to clean up a bit.